20th/articles/rfc223.md

```diff
@@ -0,0 +1,126 @@
+One of his more interesting proposals came in RFC223 on Superpositions, which suggested making his Perl **Quantum::Superposition**'s features available in the core of the language.
+
+## What is a Superposition?¹
+
+In the quantum world, there are measurable things that can exist in multiple states — simultaneously — until the point in time in which we measure them.
+For computer scientists, perhaps the most salient application of this is in qubits which, as a core element of quantum computing, threaten to destroy encryption as we know it, if quantum supremacy is borne out.
+
+At the end of the day, though, for us it means being able to treat multiple values as if they were a single value so long as never actually *need* there to only be one, at which point we get a single state from them.
+
+## The Perl Implementation
+
+In the original implementation, Dr. Conway adds two new operators, `all` and `any`.
+These converted a list of values into a *single scalar value*.
+How was this different from using a list or array?  Consider the following Perl/Raku code:
+
+    my @foo = (0, 1, 2, 3, 4, 5);
+
+We can easily access each of the values by using array notation:
+
+    print @names[0]; # 0
+    print @names[1]; # 1
+    print @names[2]; # 2
+   
+But what if we wanted to do stuff to this list of numbers?  That's a bit trickier.
+Functional programmers would probably say "But you have map!".  That's true, of course.
+If I wanted to double everything, I could say
+
+    @foo = map { $_ * 2}  @foo; # Perl
+    @foo = map { $_ * 2}, @foo; # Raku
+   
+But it could also be nice if I could just say
+
+    @foo *= 2;
+    
+This is where the superposition can be helpful.  Now imagine we have another array and wanted to add it to our now doubled set of values in `@foo`
+
+    my @bar = (0,20,40,60,80,100);
+    @foobar = @foo + @bar;          # (12); wait what?  Recall that arrays in numeric context are the number of elements, or 6 here.
+    
+Your instinctive reaction might be to say that we'd want to end up with `(0,22,44,66,88,110)` which is simple enough to handle in a basic `map` or `for` loop (using the zip operator, Raku can do this simply as `@foo Z+ @bar`).
+But remember what a superposition means: anything done happens to all the values, so each value in @foo needs to be handle with each value in @bar, which requires at least two loops if done via `map` or `for` (the cross operator in Raku can do this simply as `@foo X+ @bar`).
+We actually want `(0, 2, 4, 6, 8, 10, 20, 22, 24, 26, 28, 30, 40, 42, 44, 46, 48, 50 … )`.  More difficult, then, would be to somehow compare this value:
+
+    @foobar > 10;
+    
+There is no map method we can attach to `@foobar` to check its values against 10, we'd need to instead map the `> 10` into `@foobar`.
+But by using superpositioning, we can painless do all of the above with a single use of `map`, `for, or anything else that generates linenoise:
```

Comment:
Missed the closing `.

Reply:
This was probably fixed when uploading it, so thanks.

Reply:
Indeed I did fix it.  @Altai-man ++